includes/class-send-list.php

```
@@ -0,0 +1,213 @@
+					'required' => true,
+				],
+				'label'       => [
+					'name'     => 'label',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// If the list is also a Subscription List, it could have a locally edited name.
+				'local_name'  => [
+					'name'     => 'local_name',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// If available, the number of contacts associated with this list or sublist.
+				'count'       => [
+					'name'     => 'count',
+					'type'     => 'integer',
+					'required' => false,
+				],
+				// If this Send_List is a sublist, this property must indicate the ID of the parent list.
+				'parent_id'   => [
+					'name'     => 'parent_id',
+					'type'     => 'string',
+					'required' => false,
+				],
+				// If it can be calculated, the URL to view this list or sublist in the ESP's dashboard.
+				'edit_link'   => [
+					'name'     => 'edit_link',
+					'type'     => 'string',
+					'required' => false,
+				],
+			],
+		];
+	}
+
+	/**
+	 * Get the Send_List configuration.
+	 *
+	 * @return array
+	 */
+	public function get_config() {
+		$schema = self::get_config_schema();
+		$config  = [];
+		foreach ( $schema['properties'] as $key => $property ) {
+			$config[ $key ] = $this->get( $key ) ?? null;
+		}
+
+		return array_filter( $config );
+	}
+
+	/**
+	 * Get a specific property's value.
+	 *
+	 * @param string $key The property to get.
+	 *
+	 * @return mixed The property value or null if not set/not a supported property.
+	 */
+	public function get( $key ) {
+		return $this->{ $key } ?? null;
+	}
+
+	/**
+	 * Set a property's value.
+	 *
+	 * @param string $key The property to get.
+	 * @param mixed  $value The value to set.
+	 *
+	 * @return mixed The property value or null if not set/not a supported property.
+	 */
+	public function set( $key, $value ) {
+		$schema = $this->get_config_schema();
+		if ( ! isset( $schema['properties'][ $key ] ) ) {
+			return null;
```

We might want to return an error here.

---

Done in 7f4906e!